ppa/archive/models.py

```diff
@@ -283,13 +361,16 @@ def populate_from_bibdata(self, bibdata):
         '''
         # store hathi record id
         self.record_id = bibdata.record_id
+        print(bool(bibdata.marcxml))
 
         # set fields from marc if available, since it has more details
         if bibdata.marcxml:
+
             # set title and subtitle from marc if possible
             # - clean title: strip trailing space & slash and initial bracket
-            self.title = bibdata.marcxml['245']['a'].rstrip(' /') \
-                .lstrip('[')
+            if not self.protected_fields.title:
```

Comment:
I wonder if it would be cleaner to handle all the protected field logic in one place? Something like this: collect all the new bibliographic data in a dictionary, and then iterate over the dictionary data and set it on `self` unless the field is flagged as protected. 

I think it would make the code more readable and more efficient - handle the marc logic separate from the protected field logic.

In fact, it might be useful for the protected field logic to be in a separate method - that would make it possible to honor the protected flags if we do any other kind of bulk updates from scripts or console.

Reply:
I checked the code climate report, and it's complaining about the repeated/similar code where you're checking for protected fields, so refactoring would resolve that also.

Reply:
This worked out well. I was able to abstract the check out to a nice, short loop.